src/mdlpp/structs/MDL.cpp

```diff
@@ -66,6 +104,89 @@ bool MDL::open(const std::byte* data, std::size_t size) {
 	const auto bodyPartOffset = stream.read<int32_t>();
 
+	const auto attachmentCount = stream.read<int32_t>();
+	const auto attachmentOffset = stream.read<int32_t>();
+
+	const auto localNodeCount = stream.read<int32_t>();
+	const auto localNodeIndex = stream.read<int32_t>();
+	const auto localNodeNameIndex = stream.read<int32_t>();
+
+	const auto flexDescCount = stream.read<int32_t>();
+	const auto flexDescIndex = stream.read<int32_t>();
+	const auto flexControllerCount = stream.read<int32_t>();
+	const auto flexControllerIndex = stream.read<int32_t>();
+	const auto flexRulesCount = stream.read<int32_t>();
+	const auto flexRulesIndex = stream.read<int32_t>();
+	const auto ikChainCount = stream.read<int32_t>();
+	const auto ikChainIndex = stream.read<int32_t>();
+
+	const auto mouthsCount = stream.read<int32_t>();
+	const auto mouthsIndex = stream.read<int32_t>();
+
+	const auto localPoseParamCount = stream.read<int32_t>();
+	const auto localPoseParamIndex = stream.read<int32_t>();
+
+	const auto surfacePropertyIndex = stream.read<int32_t>();
+	const auto keyValueIndex = stream.read<int32_t>();
+	const auto keyValueCount = stream.read<int32_t>();
+
+	const auto localIKAutoplayLockCount = stream.read<int32_t>();
+	const auto localIKAutoplayLockIndex = stream.read<int32_t>();
+
+	stream
+		.read(this->mass)
+		.read(this->contentsFlags);
+
+	const auto includeModelCount = stream.read<int32_t>();
+	const auto includeModelIndex = stream.read<int32_t>();
+
+	stream.read(this->virtualModel);
+
+	// do we want to parse these here?
+	const auto animationBlocksNameIndex = stream.read<int32_t>();
+	const auto animationBlocksCount = stream.read<int32_t>();
+	const auto animationBlocksIndex = stream.read<int32_t>();
+
+	stream.read(this->animationBlockModel);
+
+	const auto boneTableNameIndex = stream.read<int32_t>();
+
+	stream
+		.read(this->vertexBase)
+		.read(this->offsetBase)
+		.read(this->directionalDotProduct)
+		.read(this->rootLOD)
+		.read(this->numAllowedRootLODs)
+		.skip<uint8_t>() // _unused0
+		.skip<int32_t>(); // _unused1
+
+	const auto flexControllerUICount = stream.read<int32_t>();
+	const auto flexControllerUIIndex = stream.read<int32_t>();
+
+	stream
+		.read(this->vertAnimFixedPointScale)
+		.skip<int32_t>() // _unused2
+		.read(this->studiohdr2index)
+		.skip<int32_t>(); // _unused3
+
+	// header of the second kind perchance
+	if (this->studiohdr2index != 0) {
+		stream.seek_u(this->studiohdr2index);
+		// TODO: Parse this
+		stream
+			.read(this->header2.srcBoneTransformCount)
+			.read(this->header2.srcBoneTransformIndex)
+			.read(this->header2.illumPositionAttachmentIndex)
+			.read(this->header2.maxEyeDeflection)
+			.read(this->header2.linearBoneIndex)
+			.read(this->header2.nameIndex)
+			.read(this->header2.boneFlexDriverCount)
+			.read(this->header2.boneFlexDriverIndex);
+		// skip the 56 reserved int32 fields
+		stream.skip<int32_t>(56);
+		this->hasHeader2 = true;
+	}
+
```

Comment:
After this point, reading in the various blocks should be in order relative to how they're ordered in the header. Not for any technical reason but because that's how I and probably other people looking at the code would expect it